src/cli_config.rs

```diff
@@ -189,6 +201,13 @@ pub fn cli_parse() -> CliConfig {
         .arg(Arg::with_name("no-lan")
             .long("no-lan")
             .help("Disable LAN discovery"))
+        .arg(Arg::with_name("config")
+            .short("c")
+            .long("config")
+            .help("Path to the file where configs are stored")
+            .takes_value(true)
+            .conflicts_with_all(&["tcp-address", "udp-address", "keys-file", "secret-key", "bootstrap-node", "threads", "log-type", "motd", "no-lan"])
```

Comment:
Better make `--config` be unique arg artificially, check the occurrence after `.get_matches`.